### PR TITLE
Remove race condition form update snippet run

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Update Snippet Generator Version.
- Fix race conditoin that occues when runing `UpdateSourceOnBuild`

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/Directory.Build.props [hunk 1]
```diff
@@ -12,6 +12,10 @@
     <IsTestSupportProject>true</IsTestSupportProject>
   </PropertyGroup>
 
+  <PropertyGroup>
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/Directory.Build.props:15]: @chidozieononiwu:  Can you help me understand the purpose of this and why we're applying it to some services and not others?

File: sdk/keyvault/Azure.Security.KeyVault.Administration/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    Add any shared properties you want for the projects under this package directory that need to be set before the auto imported Directory.Build.props
+  -->
+
+  <PropertyGroup>
```
Comment (heaths) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/Directory.Build.props:6]: Why not put this into the sdk/keyvault/Directory.Build.props to apply to all 4 shipping track 2 packages?

File: eng/Directory.Build.Common.targets [hunk 3]
```diff
@@ -39,11 +39,15 @@
...
+  <Target Name="UpdateSourceOnBuild" AfterTargets="Build" Condition="'$(UpdateSourceOnBuild)' == 'true' and '$(IsShippingClientLibrary)' == 'true' and '$(RunCodeChecksWithThisProject)' == 'true'" >
     <PropertyGroup>
       <CodeChecksScriptPath Condition=" '$(CodeChecksScriptPath)'=='' ">$(MSBuildThisFileDirectory)scripts/CodeChecks.ps1</CodeChecksScriptPath>
     </PropertyGroup>
-    <Exec ConsoleToMSBuild="true" Command="$(PowerShellExe) -NoProfile -NonInteractive -executionpolicy Unrestricted -File $(CodeChecksScriptPath) -ProjectDirectory $(MSBuildProjectDirectory)"/>
+    <PropertyGroup Condition="'$(ServiceDirectory)' == ''">
+      <ServiceDirectoryPath>$([MSBuild]::NormalizeDirectory($(MSBuildProjectDirectory)/../../))</ServiceDirectoryPath>
+      <ServiceDirectory>$([System.IO.Directory]::GetParent($(ServiceDirectoryPath)).Name)</ServiceDirectory>
+    </PropertyGroup>
+    <Exec ConsoleToMSBuild="true" Command="$(PowerShellExe) -NoProfile -NonInteractive -executionpolicy Unrestricted -File $(CodeChecksScriptPath) -ServiceDirectory $(ServiceDirectory)"/>
```
Comment (heaths) [inline eng/Directory.Build.Common.targets:50]: Tip: unlikely to have spaces, but just to protect against it: 
```suggestion
    <Exec ConsoleToMSBuild="true" Command="$(PowerShellExe) -NoProfile -NonInteractive -executionpolicy Unrestricted -File &quot;$(CodeChecksScriptPath)&quot; -ServiceDirectory &quot;$(ServiceDirectory)&quot;"/>
```

File: eng/scripts/CodeChecks.ps1 [hunk 4]
```diff
@@ -46,42 +44,32 @@ function Invoke-Block([scriptblock]$cmd) {
...
-                & dotnet sln $_ list `
-                    | ? { $_ -ne 'Project(s)' -and $_ -ne '----------' } `
-                    | % {
-                            $proj = Join-Path $slnDir $_
-                            if (-not (Test-Path $proj)) {
-                                LogError "Missing project. Solution references a project which does not exist: $proj. [$sln] "
-                            }
+    Write-Host "`nChecking that solutions are up to date"
+    Join-Path "$PSScriptRoot/../../sdk" $ServiceDirectory  `
+        | Resolve-Path `
```
Comment (heaths) [inline eng/scripts/CodeChecks.ps1:54]: Tip: you can use `Join-Path ... ... -Resolve` to accomplish the same thing.

File: sdk/core/Azure.Core/Directory.Build.props [hunk 5]
```diff
@@ -2,5 +2,10 @@
   <!--
     Add any shared properties you want for the projects under this package directory that need to be set before the auto imported Directory.Build.props
   -->
+
+  <PropertyGroup>
+    <RunCodeChecksWithThisProject>true</RunCodeChecksWithThisProject>
```
Comment (heaths) [inline sdk/core/Azure.Core/Directory.Build.props:7]: Not just for Key Vault, but seems all of these should go in sdk/<ServiceDirectory>/Directory.Build.props. Track 1 targets wouldn't use it, so it doesn't hurt anything there. And all track 2 projects will get it. Many - perhaps even most - service directories have multiple track 2 projects.

**Review by heaths — CHANGES_REQUESTED** [review]

**Pull request closed without merging**